tests/unit/helpers/common-helper.test.ts

```diff
@@ -92,3 +93,38 @@ describe("Pluralize a word based on the count", () => {
         expect(pluralize(0, "item", "items")).toBe("items");
     });
 });
+
+describe("Group items of a list by a key (function)", () => {
+    it("When list is empty,then no items are grouped", () => {
+        expect(groupBy([], (i) => i)).toStrictEqual({});
+    });
+
+    it("When the key selector function is given, then the items are grouped by their corresponding key determined by the key selector function", () => {
+        const list = [
+            { type: "a", value: 1 },
+            { type: "a", value: 2 },
+            { type: "b", value: 3 },
+            { type: "a", value: 1 },
+        ];
+        expect(groupBy(list, (i) => i.type)).toStrictEqual({
+            a: [
+                { type: "a", value: 1 },
+                { type: "a", value: 2 },
+                { type: "a", value: 1 },
+            ],
+            b: [{ type: "b", value: 3 }],
+        });
+        expect(groupBy(list, (i) => "" + i.value)).toStrictEqual({
```

Comment:
`"" + i.value` looks like there should be something inside "".
What about
```suggestion
        expect(groupBy(list, (i) => `${i.value}`)).toStrictEqual({
```
or `i.value.toString()`

Reply:
Yes I agree, that this cast isn't the best solution, I changed it to your first suggestion.